runtime/autoload/macvim.vim

```diff
@@ -76,4 +76,41 @@ export def ShowDefinitionUnderCursor()
   endif
 enddef
 
+# Print functionality. We simply show the file in Preview and let the user
+# decide what to do. This allows for more control instead of immediately
+# piping the file to lpr which will actually print the file.
+#
+# PreviewConvertPostScript: 
+#   Convert the provided PostScript file to PDF, then show in Preview. This is
+#   necessary in macOS 13+ as Preview doesn't support .ps files anymore.
+# PreviewPostScript:
+#   Directly open PostScript file in Preview. Can use this if
+#   PreviewConvertPostScript doesn't work.
+export def PreviewConvertPostScript(deltimer = 10000): number
+  # Convert PS to PDF because Preview can't use PS files in macOS 13+
+  system($"pstopdf {v:fname_in} -o {v:fname_in}.pdf")
+  if v:shell_error != 0
+    return v:shell_error
+  endif
+  system($"open -a Preview {v:fname_in}.pdf")
+  delete(v:fname_in)
+
+  # Delete the file after it's opened in Preview for privacy. We don't have an
+  # easy way to detect that Preview has opened the file already (unless we use
+  # `open -W` which will stall Vim), so we just use a generous 10 secs timer.
```

Comment:
The main issue is actually that `-W` only terminates when Preview.app closes. A normal user may have multiple documents opened in Preview, which never closes, meaning that `-W` has to wait forever to finish. You can use some AppleScript to figure out when the document has closed but it's somewhat convoluted and also requires asking permissions from the user.

Otherwise we could have simply used `call system(open -W Preview.app; rm ... &)` to open it in the background.

Reply:
I'll just remove that comment about -W.

Reply:
I agree, my idea was to call an external routine(script or via client-server architectrure) which will fork and wait and remove files as they not needed

Reply:
Please, keep the comment, it's useful

Reply:
Hmm, I can add something back. The key point here though is that `-W` does not do what we want, so the original comment was wrong. It only works if the user literally only has one file open and immediately close Preview, which is often not the case.

Reply:
Makes sense, thanks